boards: arm: Add support for STM NUCLEO-H753ZI

JeremyBCD opened the pull request:
Based on NUCLEO-H743ZI. NUCLEO-H753ZI adds support for hardware crypto and secure boot.

Also added `CONFIG_SOC_STM32H753XX` to `drivers/clock_control/clock_stm32_ll_h7.c`.

JeremyBCD committed:
boards: arm: Add nucleo_h753zi, based on nucleo_h743zi

* Added boards/arm/nucleo_h753zi, based on boards/arm/nucleo_h743zi.

Signed-off-by: Jeremy Wood <jeremy@bcdevices.com>

JeremyBCD commented:
I'm unable to resolve the style issue in the compliance check, the change matches the style in that file, but not the project style guide.

JeremyBCD commented:
Fixed `boards/arm/nucleo_h753zi/support/openocd.cfg`, using `board/st_nucleo_h743zi.cfg`.

FRASTM reviewed (approved):
maybe "Copyright (c) 202**1**" ?

JeremyBCD commented:
@FRASTM Good point—updated copyright.

erwango commented on boards/arm/nucleo_h753zi/Kconfig.board:
```diff
@@ -0,0 +1,9 @@
+# STM32H753ZI Nucleo board configuration
+
+# Copyright (c) 2020 Teslabs Engineering S.L.
```
Please keep only your copyright

erwango commented on boards/arm/nucleo_h753zi/Kconfig.defconfig:
```diff
@@ -0,0 +1,19 @@
+# STM32H753ZI Nucleo board configuration
+
+# Copyright (c) 2020 Teslabs Engineering S.L.
```
Please keep only your copyright

erwango commented on boards/arm/nucleo_h753zi/arduino_r3_connector.dtsi:
```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2019 Linaro Limited
+ * Copyright (c) 2020 Teslabs Engineering S.L.
```
Please keep only your copyright

JeremyBCD committed:
drivers: clock_control: Add support for STM32H753XX.

Added support for STM32H753XX by adding CONFIG_SOC_STM32H753XX to list
of H7 SoC with maximum 480MHz SYSCLK.

Signed-off-by: Jeremy Wood <jeremy@bcdevices.com>

JeremyBCD committed:
tests: adc: Add support for NUCLEO_H753ZI.

Add CONFIG_BOARD_NUCLEO_H753ZI to ADC device test definition.

Signed-off-by: Jeremy Wood <jeremy@bcdevices.com>

JeremyBCD commented:
@erwango Thanks for the review—I've removed the extraneous copyright, and moved the test and driver changes into separate commits.

erwango commented on drivers/clock_control/clock_stm32_ll_h7.c:
```diff
@@ -1,7 +1,6 @@
  *
- * Copyright (c) 2019 Linaro Limited.
- * Copyright (c) 2020 Jeremy LOCHE
+ * Copyright (c) 2021 Blue Clover Devices
```
I might be the one to blame here:
Please keep existing copyrights here (you're not adding a new file)

JeremyBCD replied:
That's my bad, I should have realized that from the contribution guidelines.

Fixed here and in the adc tests.

erwango reviewed (changes requested):
Thanks for the changes.
One final tweak and we're good to go.
